### PR TITLE
doc: add various docstrings

### PR DESCRIPTION
**Pull request by EdAyers**
more to come

### CODE REVIEW

File: src/Lean/LocalContext.lean [hunk 1]
```diff
@@ -10,7 +10,9 @@ import Lean.Hygiene
 namespace Lean
 
 inductive LocalDecl where
+  /-- A local constant. -/
```
[the remark this answers is not in the document]
Comment (leodemoura) [reply src/Lean/LocalContext.lean:13]: The doc-string should be after the `|` and before the identifier.

File: src/Init/Meta.lean [hunk 2]
```diff
@@ -987,6 +987,13 @@ end Syntax
 
 namespace Meta
 
+/-- A parameter representing how aggressively definitions should be unfolded when trying to decide if two terms match, unify or are definitionally equal.
+By default, theorem declarations are never unfolded.
+- `all` will unfold everything, including irreducible definitions and theorems
+- `default` will unfold everything except theorems and definitions tagged as irreducible.
```
Comment (leodemoura) [inline src/Init/Meta.lean:993]: and theorems.

**Comment by leodemoura** [comment]
@EdAyers Thanks for doing this.

File: src/Lean/LocalContext.lean (hunk 1 above)
[the remark this answers is not in the document]
Comment (EdAyers) [reply src/Lean/LocalContext.lean]: Like `| /-- A local variable -/ cdecl (index : Nat) ...`?

[the remark this answers is not in the document]
Comment (leodemoura) [reply src/Lean/LocalContext.lean]: Yes. Here is an example where we use it https://github.com/leanprover/lean4/blob/master/src/Lean/Meta/CongrTheorems.lean#L10-L33

**Comment by leodemoura** [comment]
@EdAyers Thanks again for doing this. It seems you are full of energy and wants to add many docstrings. If this is the case, I think we can use the following suggestions to get your PRs merged as quickly as possible.
 
- Several small PRs are better than a big one. Feel free to create one per file. 
- In case of doubt, ping us on the PR. No docstring is better than an incorrect one.
- Focus on "stable" code. All docstrings in this PR are on "stable" code, and this is great. The git blame is great to check whether a code has not changed in the last few months. 
- Focus on the functions that are most useful to the community. It seems you are already doing this.

**Comment by EdAyers** [comment]
ok, I'll close this PR and split in to multiple files

**Pull request closed without merging**